### PR TITLE
#26637: Fixing issue with Invalid customer id

### LINKED ISSUE #23618: Invalid address id at check out
### Preconditions (*)

1. Magento 2.3.2
2.

### Steps to reproduce (*)

1. Delete a customer and then re-create it with the same email.
2. Go to checkout with a product in the basket.
3. At sending press "next".

### Expected result (*)

1. The next step is supposed to load

### Actual result (*)

1. I get the error:  Unable to save shipping information. Please check input data.
2. Error log: [2019-07-08 10:33:00] main.CRITICAL: Invalid customer-address id 6 {"exception":"[object] (Magento\\Framework\\Exception\\NoSuchEntityException(code: 0): Invalid customer-address id 6 at /home/server/domains/dev3.domain.com/public_html/vendor/magento/module-quote/Model/QuoteAddressValidator.php:79)"} []

3. I removed the address from this customer and re-entered it. Ran the index etc. Nothing.
4. Other accounts are still ok with the checkout.
5. The table quote_address stays empty for this quote.

### PR DESCRIPTION
**Pull request by srenon**
### Description (*)

Invalid customer ID issue coming on the checkout while placing the order. This issue comes randomly for the customers who are logged in on the website. This issue is related to the customer quote, the table quote_address stays empty for this quote.

### Fixed Issues (if relevant)

1. Fixes magento/magento2#23618 : Invalid address id at check out

### Manual testing scenarios (*)

1. Disable guest checkout
2. While still logged out, add an item to your cart (has this issue with downloadable product)
3. Then click checkout
4. On the log in screen. Log in with a customer that already has a save cart and billing info

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

File: app/code/Magento/Quote/Model/QuoteManagement.php [hunk 1]
```diff
@@ -494,6 +494,12 @@ protected function resolveItems(QuoteEntity $quote)
      */
     protected function submitQuote(QuoteEntity $quote, $orderData = [])
     {
+        if ($quote->getCustomer()->getId()) {
+            $quote->setCustomerIsGuest(0);
+        } else {
+            $quote->setCustomerIsGuest(1);
+        }
```
Comment (ihor-sviziev) [inline app/code/Magento/Quote/Model/QuoteManagement.php]:
```suggestion
        $quote->setCustomerIsGuest((int)!$quote->getCustomer()->getId());
```

**Review by ihor-sviziev — CHANGES_REQUESTED** [review]
Hi @srenon,
Could you do following?
- apply suggested change with simplification 
- add some data upgrade script that will update all quotes with incorrect data in DB? (see example https://github.com/magento/magento2/issues/23908#issuecomment-610248165)
- cover your change with any type of automated tests?

**Comment by ihor-sviziev** [comment]
@magento run all tests

**Comment by ihor-sviziev** [comment]
Seems like this PR also fixes #23908

**Comment by engcom-Hotel** [comment]
@magento run all tests

File: app/code/Magento/Quote/Model/QuoteManagement.php (hunk 1 above)
Comment (engcom-Hotel) [reply app/code/Magento/Quote/Model/QuoteManagement.php]: Has been updated.

**Comment by ihor-sviziev** [comment]
@engcom-Hotel tests still failing, could you check it ?

**Comment by engcom-Hotel** [comment]
@ihor-sviziev will check it.

**Comment by ihor-sviziev** [comment]
@magento run all tests

**Review by ihor-sviziev — CHANGES_REQUESTED** [review]
Moving to correct status. Tests should be adjusted.

**Comment by engcom-Hotel** [comment]
Hello @srenon
Could you please provide more detailed steps to reproduce the issue?
Thank you.

**Comment by engcom-Hotel** [comment]
@magento run all tests

**Comment by engcom-Hotel** [comment]
Hello @srenon.
Please respond to my previous comment. Otherwise, PR could be closed due to inactivity.
Thank you.

**Comment by JKetelaar** [comment]
Still facing this issue, though with this PR it's fixed - so please do not close it...

**Comment by engcom-Hotel** [comment]
@JKetelaar could you please provide detailed steps to reproduce the issue?
Thank you.

**Comment by engcom-Hotel** [comment]
Hello @srenon,
Closing the issue due to inactivity.
Please feel free to reopen it.
Thanks.

**Comment by groomershop-mt** [comment]
In our case this commits also don't  fix problem at Magento 2.4.2.

**Comment by barryvdh** [comment]
@groomershop-mt  did you apply this patch?

**Pull request closed without merging**